File: js/aa/isPalindrome.js
function isPalindrome(str){
  const len = str.length;
  for (let i = 0; i < Math.floor(len / 2); i++) {
    if (str[i] !== str[len - 1 - i]){
      return false;
    }
  }

  return true;
}

console.log(isPalindrome("abcdedcba"));
console.log(isPalindrome("abcdeedcba"));
console.log(isPalindrome("asldghdlfj"));

File: ruby/aa/is_palindrome.rb
def is_palindrome(str)
  len = str.length
  (0...len / 2).each do |i|
    return false if str[i] != str[len - 1 - i]
  end

  true
end

p is_palindrome("abcdedcba")
p is_palindrome("abcdeedcba")
p is_palindrome("asldghdl")
